### ISSUE TITLE
Start a program in user folder leaves file handle open in SYSTEM

### ISSUE THREAD

**Issue by yume-chan** [PowerShell/PowerShell#14592]
## Steps to reproduce

Requires an executable in user folder, for example, I have `yarn` installed by `npm` at `C:\Users\chenximin\AppData\Roaming\npm`

1. Start `cmd.exe`
2. Run `yarn -v` in cmd
4. Search for `yarn.cmd` in Process Explorer, found nothing
    ![image](https://user-images.githubusercontent.com/1330321/104145667-de3adf00-5402-11eb-8616-a536326f28db.png)
5. Start `powershell.exe`
6. Run `yarn -v` in PowerShell 5
7. Search for `yarn.ps1` in Process Explorer again, found nothing
   ![image](https://user-images.githubusercontent.com/1330321/104145592-91570880-5402-11eb-806d-2cd9e230d627.png)
8. Start `pwsh.exe`
9. Run `yarn -v` in PowerShell Core
10. Search for `yarn.ps1` in Process Explorer again

## Expected behavior

Also found nothing

## Actual behavior

Found a file handle opened in SYSTEM, so I can't delete this file

![image](https://user-images.githubusercontent.com/1330321/104145604-a0d65180-5402-11eb-89d5-1f77c0cdbb0f.png)

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.2.0-preview.2
PSEdition                      Core
GitCommitId                    7.2.0-preview.2
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

I don't have any anti-virus software, don't have any file sharing.

**Comment by iSazonov** [comment]
Is it the file https://github.com/yarnpkg/yarn/blob/3119382885ea373d3c13d6a846de743eca8c914b/bin/yarn.ps1#L1-L6 ? If so there node is started and I guess we could find more simple reproduce steps.

**Comment by yume-chan** [comment]
No, `yarn` is just an example, I have problem with all executables in my user folder, usually it's `code-insiders.cmd` and `yarn.ps1`.

After more researching I found that it only happens to PowerShell installed from Microsoft Store. I downloaded the .msi installer from GitHub releases and that one is fine. Maybe related to my old Windows 10 version and Microsoft Store.

**Comment by iSazonov** [comment]
> I found that it only happens to PowerShell installed from Microsoft Store

Thanks for your investigations!

/cc @SteveL-MSFT for information.
